libs/ledger-live-common/src/onboarding/hooks/useOnboardingStatePolling.ts

```diff
@@ -0,0 +1,246 @@
+                return of(error);
+              }
+
+              console.log(
+                `SyncOnboarding: 💥 Fatal Error ${error} -> ${JSON.stringify(
+                  error
+                )}`
+              );
+              return throwError(error);
+            }),
+            map((deviceVersionOrAllowedError: FirmwareInfo | Error) => {
+              // TODO: better safe guard function ?
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              if (deviceVersionOrAllowedError?.flags) {
+                console.log(
+                  `SyncOnboarding: ♧ MAP got firmwareInfo: ${JSON.stringify(
+                    deviceVersionOrAllowedError
+                  )}`
+                );
+
+                let onboardingState: OnboardingState | null = null;
+
+                try {
+                  onboardingState = extractOnboardingState(
+                    (deviceVersionOrAllowedError as FirmwareInfo).flags
+                  );
+                } catch (error) {
+                  console.log(
+                    `SyncOnboarding: extract onboarding error ${JSON.stringify(
+                      error
+                    )}`
+                  );
+                  if (error instanceof DeviceExtractOnboardingStateError) {
+                    return {
+                      onboardingState: null,
+                      allowedError:
+                        error as typeof DeviceExtractOnboardingStateError,
+                    };
+                  } else {
+                    return {
+                      onboardingState: null,
+                      allowedError: new DeviceOnboardingStatePollingError(
+                        "SyncOnboarding: Unknown error while extracting the onboarding state"
+                      ),
+                    };
+                  }
+                }
+                return { onboardingState, allowedError: null };
+              }
+
+              console.log(
+                `SyncOnboarding: ♧ MAP got accepted error: ${JSON.stringify(
+                  deviceVersionOrAllowedError
+                )}`
+              );
+              return {
+                onboardingState: null,
+                allowedError: deviceVersionOrAllowedError as Error,
+              };
+            })
+          );
+
+      // Delays the fetch of the onboarding state
+      setTimeout(() => {
+        getOnboardingStateOnce().subscribe({
+          next: (value: OnboardingStatePollingResult) => {
+            subscriber.next(value);
+          },
+          error: (error: any) => {
+            subscriber.error(error);
+          },
+          // Import for repeat()
+          complete: () => subscriber.complete(),
+        });
+      }, delayMs);
+    });
+
+  return getDelayedOnboardingStateOnce.pipe(repeat());
+};
+
+// TODO: decide which errors are allowed
+export const isAllowedOnboardingStatePollingError = (
+  error: Error | any
```

Comment:
Not sure why it's error | any here. But I'd say that this type of function should only accept Error, and if it's not an error it's up to the caller to check before passing it down to the function.

Reply:
agreed, done ;)